packages/ui-drawer-layout/src/DrawerLayout/DrawerTray/props.ts

```diff
@@ -35,38 +35,132 @@ import type {
   PropValidators,
   DrawerLayoutTrayTheme,
   OtherHTMLAttributes,
-  LiveRegion
+  LiveRegion,
+  UIElement
 } from '@instructure/shared-types'
+import {
+  BaseTransitionStatesType,
+  TransitionType
+} from '@instructure/ui-motion/src/Transition/BaseTransition/props'
 
 type DrawerTrayPlacement = 'start' | 'end'
 
 type DrawerLayoutTrayOwnProps = {
   label: string
-  children?: React.ReactNode | ((...args: any[]) => React.ReactNode)
-  render?: (...args: any[]) => any
+
+  children?: React.ReactNode | (() => React.ReactNode)
+
+  render?: () => React.ReactNode
+
+  /**
+   * Placement of the `<DrawerLayout.Tray />`
+   */
   placement?: DrawerTrayPlacement
+
+  /**
+   * If the tray is open or closed.
+   */
   open?: boolean
-  onOpen?: (...args: any[]) => any
-  onClose?: (...args: any[]) => any
+
+  /**
+   * Called when the `<DrawerLayout.Tray />` is opened
+   */
+  onOpen?: (transitionType?: TransitionType) => void
+
+  /**
+   * Called when the `<DrawerLayout.Tray />` is closed
+   */
+  onClose?: (transitionType?: TransitionType) => void
```

Comment:
These args are not necessarily needed here but the Transition component provides them anyway.